### PR TITLE
chore/ci: github actions for release process

### PR DESCRIPTION
**Pull request by m-cat**
Closes https://github.com/maidsafe/QA/issues/110

See the safe-cli PR as a reference: https://github.com/maidsafe/safe-api/pull/330

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -121,152 +63,39 @@ endif
...
 	rm -f *.tar
 	rm -rf deploy
-	mkdir deploy
-	tar -C artifacts/linux/release -cvf safe_vault-$$(git rev-parse --short HEAD)-x86_64-unknown-linux-musl.tar safe_vault
-	tar -C artifacts/win/release -cvf safe_vault-$$(git rev-parse --short HEAD)-x86_64-pc-windows-gnu.tar safe_vault.exe
-	tar -C artifacts/macos/release -cvf safe_vault-$$(git rev-parse --short HEAD)-x86_64-apple-darwin.tar safe_vault
-	mv safe_vault-$$(git rev-parse --short HEAD)-x86_64-unknown-linux-musl.tar deploy
-	mv safe_vault-$$(git rev-parse --short HEAD)-x86_64-pc-windows-gnu.tar deploy
-	mv safe_vault-$$(git rev-parse --short HEAD)-x86_64-apple-darwin.tar deploy
+	mkdir -p deploy/prod
```
Comment (lionel1704) [inline Makefile]: Changing this path _might_ impact auto-update.
Could you please make sure that auto-update is unaffected by this change?

Comment (m-cat) [reply Makefile]: Looks like self_update just gets the latest release on GitHub. I've already tested the new release process with these changes and confirmed it works.

I've also searched for "deploy" and confirmed this path was not being referred to anywhere else.

**Comment by lionel1704** [comment]
@m-cat do we still need the `Jenkinsfile` in this repo?

**Comment by m-cat** [comment]
Nope, GHA supersedes it. Once this is merged we can decommission the Jenkins slaves ASAP 🙂

**Pull request merged**